libs/digger_config/digger_config.go

```diff
@@ -269,15 +269,23 @@ func HandleYamlProjectGeneration(config *DiggerConfigYaml, terraformDir string,
 							workflow = b.Workflow
 						}
 
-						err := hydrateDiggerConfigYamlWithTerragrunt(config, TerragruntParsingConfig{
+						tgParsingConfig := TerragruntParsingConfig{
 							CreateProjectName: true,
 							DefaultWorkflow:   workflow,
 							WorkflowFile:      b.WorkflowFile,
 							FilterPath:        path.Join(terraformDir, *b.RootDir),
-						}, terraformDir)
+						};						
+
+						// allow blocks to pass in roles that can be assummed by aws 
+						if b.AwsRoleToAssume != nil {
```

Comment:
nitpick:  I think its safe to pass b.AwsRoleToAssume directly into the struct as it will be passed into the project as nil and handled well in downstream functions

Reply:
Yep, you are right will adjust.